net/net.go

```diff
@@ -234,12 +234,37 @@ func (n *net) CreateThread(_ context.Context, id thread.ID, opts ...core.NewThre
 	return n.getThreadWithAddrs(id)
 }
 
-func (n *net) ensureUnique(id thread.ID) error {
-	_, err := n.store.GetThread(id)
-	if err == nil {
-		return lstore.ErrThreadExists
+func (n *net) ensureUnique(id thread.ID, key crypto.Key) error {
+	// Check if thread exists.
+	thrd, err := n.store.GetThread(id)
+	if errors.Is(err, lstore.ErrThreadNotFound) {
+		return nil
 	}
-	if !errors.Is(err, lstore.ErrThreadNotFound) {
+	if err != nil {
+		return err
+	}
+	// Early out if no log key required.
```

Comment:
If we don't care about log keys, then we'll end up creating a random one downstream, with negligible chance of a collision.